keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -0,0 +1,239 @@
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Writing logs to files](#writing-logs-to-files)
+  - [Caveats](#caveats)
+  - [Risks and Mitigations](#risks-and-mitigations)
+    - [Users don't want to use go-runner as replacement.](#users-dont-want-to-use-go-runner-as-replacement)
+    - [Log processing in parent process causes performance problems](#log-processing-in-parent-process-causes-performance-problems)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+    - [Beta](#beta)
+    - [GA](#ga)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Continue supporting all klog features](#continue-supporting-all-klog-features)
+  - [Release klog 3.0 with removed features](#release-klog-30-with-removed-features)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes to deprecate and in the future to remove a subset of the klog
+command line flags from Kubernetes components to encourage more diverse 
+approaches to logging in kubernetes ecosystem logging.
```

Comment:
More diverse approaches? I think that would expand our support surface area, not shrink it. I think we should proceed with the rationale that removing some flags reduces support complexity and possible Kubernetes-supported configurations.

Reply:
Rewrote this part. What I meant here is similar to [Open-closed principle](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle). By  making modification harder, we make logging more simpler and easier to extend.